org-code-javabuilder/README.md

```diff
@@ -1,33 +1,138 @@
+
+**Deploying Javabuilder**  
+To deploy a development version of Javabuilder, see the
+[Deploying Javabuilder](https://docs.google.com/document/d/1mMQK6HhniLsz9lynzhUcm7Tcw_2WVLBxADe0WzqL6rM/edit#)
+instruction doc.
+
+Once you have deployed Javabuilder, you can run a basic connection to it with
+```
+wscat -c wss://<host-name>/?Authorization=connectivityTest
+```
+**Developing with deployed Javabuilder**  
+If you need to test against a level's Java code, use the following steps:
+1. Log in to https://staging-studio.code.org 
+1. Navigate to the level or project you'd like to test against
+1. Open the network tab on your browser developer tools
+1. Click "Run" in Javalab
+1. Look for the network request to 
+   https://staging-studio.code.org/javabuilder/access_token?projectUrl=...
+   in the response, there will be a token. Copy the entire token, without quotes around
+   it.
+1. Open up Postman. Choose new-> WebSocket Request 
+1. Put the url of your javabuilder instance in the server url section (such as 
+   wss://javabuilder-myname.dev-code.org)
+1. Under Params add the key `Authorization` with the value of the token you copied earlier
+1. Under Headers add the key `Origin` with the value https://staging-studio.code.org
+1. Click connect
+
+You should now start seeing messages from Javabuilder! Your token will last for 15
+minutes. If you make changes to the code, you do not need to re-generate your token.
 
-### Developing with AWS (under construction)
-In order to test against the AWS services, you'll need to integrate your IDE with AWS SAM.
+**Developing with deployed Javabuilder and an adhoc environment**  
+1. Deploy javabuilder with the instructions above
+1. In the code-dot-org repo, edit the `javabuilder_url` value in 
+   [cdo.rb](https://github.com/code-dot-org/code-dot-org/blob/3219e5866689117e086d9891effe0fb39b9ae3f0/lib/cdo.rb#L131)
+   to point to your local dev deployment.
+1. Deploy the adhoc using the instructions in the 
+   [How to Provision an adhoc Environment](https://docs.google.com/document/d/1nWeQEmEQF1B2l93JTQPyeRpLEFzCzY5NdgJ8kgprcDk/edit)
+   document.
 
-1. Set up AWS SAM: https://docs.aws.amazon.com/toolkit-for-jetbrains/latest/userguide/setup-toolkit.html
-1. Connect the first time: https://docs.aws.amazon.com/toolkit-for-jetbrains/latest/userguide/key-tasks.html#key-tasks-first-connect
-1. This is still in development. Check in with the CSA team to get the latest details on running with AWS.
-1. Open LambdaRequestHandler, set any configurations, and run the lambda.
+### Developing in isolation
+To run Javabuilder in isolation, run the `LocalMain` class from you Java IDE. This will
+exercise the compile and run logic in isolation. Javabuilder will execute the program in
+src/main/resources/main.json. You can update which file is used from 
+`LocalProjectFileManager`. Input and output will be directed to the terminal. Any 
+non-Java file used or created by the program (for example `grid.txt` in 
```

Comment:
`grid.txt` will also be created when you run neighborhood against a `gradle appRun` instance of javabuilder, not sure if that should be called out somewhere

Reply:
👍